### PR TITLE
Viash 0.7.2 (2023-04-07): Project-relative paths and improved metadata handling

### PR DESCRIPTION
**Pull request by rcannood**
This update adds functionality to resolve paths starting with a slash as relative to the project directory, improves handling of info metadata in the config, and fixes to the operator precedence of config mods.

## NEW FUNCTIONALITY

* Resolve resource and merge paths starting with a slash (`/`) as relative to the project directory (#380). To define absolute paths (which is not recommended anyway), prefix the path with the `file://` protocol. Examples:

  - `/foo` is a file or directory called `foo` in the current project directory.
  - `file:/foo` is a file or directory called `foo` in the system root.

## MINOR CHANGES

* `config yaml`: Do not modify (e.g. strip empty fields) of the `.functionality.info` and `.functionality.arguments[].info` fields (#386).

## BUG FIXES

* `ConfigMods`: Fix operator precedence issues with conditions in the config mod parsers (#390).

## INTERNAL CHANGES

* Clean up unused code (#380).

* Move circe encoders/decoders for File and Path from `io.viash.functionality.arguments` to `io.viash.helpers.circe` (#380).

* Store the project root directory (that is, the directory of the `_viash.yaml`) in a ViashProject object (#380).

* Tests: Reworked language tests to be grouped in their own subfolder and split off the bash language test from the general `testbash` folder (#381).

* Tests: Add additional language tests for `viash config inject` (#381).

* Tests: Added test for `io.viash.helpers.IO` (#380).

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name := "viash"
 
-version := "0.7.1"
+version := "0.7.3"
```
Comment (Grifs) [inline build.sbt:3]: version := "0.7.2"

**Review by Grifs — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 1 above)
Comment (rcannood) [reply build.sbt:3]: Derp, glad somebody's paying attention :p can you surround your suggestion with a suggestion code block?

Comment (Grifs) [reply build.sbt:3]: Aha, I missed how that worked and thought it picked it up by itself. Should be fixed now

Comment (rcannood) [reply build.sbt]: :+1:

**Review by Grifs — APPROVED** [review]

**Pull request merged**